src/guardian/guardian.service.ts

```diff
@@ -136,14 +137,14 @@ export class GuardianService implements OnModuleInit {
         blockHash: blockData.blockHash,
       });
 
-      await Promise.all(
-        stakingModules.map(async (stakingRouterModule) => {
-          const stakingModuleData =
-            await this.stakingModuleGuardService.getStakingRouterModuleData(
-              stakingRouterModule,
-              blockHash,
-            );
+      // TODO: check only if one of nonce changed
+      await this.stakingModuleGuardService.checkVettedKeysDuplicates(
```

Comment:
in this code, if there is a key crossing, it will stop all deposits, is this the expected behavior?

Reply:
perhaps we should reconsider here and skip deposits only in modules where there are overlaps

Reply:
you are write, i wrote it before decided to skip deposits only for one module
will fix